docs/core/extensions/queue-service.md

```diff
@@ -0,0 +1,116 @@
+---
+title: Create a Queue Service
+description: Learn how to create a queue service subclass of BackgroundService in .NET.
+author: IEvangelist
+ms.author: dapine
+ms.date: 05/26/2021
+ms.topic: tutorial
+---
+
+# Create a Queue Service
+
+A queue service is a great example of a long-running service, where work items can be queued and worked on sequentially as previous work items are completed. Relying on the Worker Service template, you'll build out new functionality on top of the <xref:Microsoft.Extensions.Hosting.BackgroundService>.
+
+In this tutorial, you learn how to:
+
+> [!div class="checklist"]
+>
+> - Create a queue service.
+> - Delegate work to a task queue.
+> - Register a console key-listener from <xref:Microsoft.Extensions.Hosting.IHostApplicationLifetime> events.
+
+## Prerequisites
+
+- The [.NET 5.0 SDK or later](https://dotnet.microsoft.com/download/dotnet)
+- A .NET integrated development environment (IDE)
+  - Feel free to use [Visual Studio](https://visualstudio.microsoft.com)
+
+<!-- ## Create a new project -->
+[!INCLUDE [file-new-worker](includes/file-new-worker.md)]
+
+## Create queuing services
+
+You may be familiar with the <xref:System.Web.Hosting.HostingEnvironment.QueueBackgroundWorkItem(System.Func{System.Threading.CancellationToken,System.Threading.Tasks.Task})> functionality from the `System.Web.Hosting` namespace. To model a service that is inspired by this functionality, start by adding an `IBackgroundTaskQueue` interface to the project:
+
+:::code source="snippets/workers/queue-service/IBackgroundTaskQueue.cs":::
+
+There are two methods, one which exposes queuing functionality and the other that dequeues previously queued work items. A *work item* is a `Func<CancellationToken, ValueTask>`. Next, add the default implementation to the project.
+
+:::code source="snippets/workers/queue-service/DefaultBackgroundTaskQueue.cs":::
+
+The preceding implementation relies on a <xref:System.Threading.Channels.Channel%601> as a queue. The <xref:System.Threading.Channels.BoundedChannelOptions.%23ctor(System.Int32)> is called with an explicit capacity. Capacity should be set based on the expected application load and number of concurrent threads accessing the queue. <xref:System.Threading.Channels.BoundedChannelFullMode.Wait?displayProperty=nameWithType> will cause calls to <xref:System.Threading.Channels.ChannelWriter%601.WriteAsync%2A?displayProperty=nameWithType> to return a task, which completes only when space became available. This leads to backpressure, in case too many publishers/calls start accumulating.
```

Comment:
```suggestion
The preceding implementation relies on a <xref:System.Threading.Channels.Channel%601> as a queue. The <xref:System.Threading.Channels.BoundedChannelOptions.%23ctor(System.Int32)> is called with an explicit capacity. Capacity should be set based on the expected application load and number of concurrent threads accessing the queue. <xref:System.Threading.Channels.BoundedChannelFullMode.Wait?displayProperty=nameWithType> will cause calls to <xref:System.Threading.Channels.ChannelWriter%601.WriteAsync%2A?displayProperty=nameWithType> to return a task, which completes only when space becomes available. This leads to backpressure, in case too many publishers/calls start accumulating.
```

TIL about backpressure :)